test/conformance/serial/ptp.go

```diff
@@ -838,6 +840,260 @@ var _ = Describe("["+strings.ToLower(DesiredMode.String())+"-serial]", Serial, f
 				})
 			})
 		})
+
+		Context("WPC GM Verification Tests", func() {
+			BeforeEach(func() {
+				By("Refreshing configuration", func() {
+					ptphelper.WaitForPtpDaemonToBeReady()
+					fullConfig = testconfig.GetFullDiscoveredConfig(pkg.PtpLinuxDaemonNamespace, true)
+				})
+				if fullConfig.Status == testconfig.DiscoveryFailureStatus {
+					Skip("Failed to find a valid ptp slave configuration")
+				}
+			})
+			//TODO: Use Focus Flag to separate GM tests from other tests
+			It("is verifying WPC GM state based on metrics", func() {
+				if fullConfig.PtpModeDiscovered != testconfig.TelcoGrandMasterClock {
+					Skip("test valid only for GM test config")
+				}
+				By("checking GM required processes status", func() {
+					/*
+						# TYPE openshift_ptp_process_status gauge
+						openshift_ptp_process_status{config="ptp4l.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="phc2sys"} 1
+						openshift_ptp_process_status{config="ptp4l.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="ptp4l"} 1
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="gpsd"} 1
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="gpspipe"}
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="gpspipe"} 1
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="ts2phc"} 1
+					*/
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring(metrics.OpenshiftPtpProcessStatus),
+						"Process status metrics are not detected")
+
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring("phc2sys"),
+						"phc2ys process status not detected")
+
+					time.Sleep(10 * time.Second)
+					buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+					ret, err := processRunning(buf.String())
+					Expect(err).To(BeNil())
+					Expect(ret["phc2sys"]).To(BeTrue(), "Expected phc2sys to be running for GM")
+					Expect(ret["gpspipe"]).To(BeTrue(), "Expected gpspipe to be running for GM")
+					Expect(ret["ts2phc"]).To(BeTrue(), "Expected ts2phc to be running for GM")
+					Expect(ret["gpsd"]).To(BeTrue(), "Expected gpsd to be running for GM")
+					Expect(ret["ptp4l"]).To(BeTrue(), "Expected ptp4l to be running for GM")
+					time.Sleep(1 * time.Minute)
```

Comment:
why sleep one minute here

Reply:
added this after discussion with @aneeshkp as the metrics may not be available for a minute or so after all the processes are available